test/v200/utils/api/test_utils.go

```diff
@@ -0,0 +1,173 @@
+package api
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"testing"
+
+	common "github.com/devfile/api/v2/test/v200/utils/common"
+	"github.com/santhosh-tekuri/jsonschema"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// numDevfiles : the number of devfiles to create for each test
+	numDevfiles = 5
+
+	schemaFileName = "../../../schemas/latest/ide-targeted/devfile.json"
+)
+
+var schemas = make(map[string]SchemaFile)
+
+type SchemaFile struct {
+	Schema *jsonschema.Schema
+}
+
+type DevfileValidator struct{}
+
+// WriteAndVerify implements Saved.DevfileValidator interface.
+// writes to disk and validates the specified devfile
+func (devfileValidator DevfileValidator) WriteAndValidate(devfile *common.TestDevfile) error {
```

Comment:
perhaps function `WriteAndValidate` and `writeDevfile` can be under pkg `test/v200/utils/common`?
So you do not need to define the empty struct `DevfileValidator` to avoid import cycle

Reply:
The method implementations are completely different comparing the api and library implementations so they do not belong in the common. This is the standard way of implementing an interface declared in a different package.